src/Elastic.Apm.AspNetFullFramework/ElasticApmModule.cs

```diff
@@ -0,0 +1,226 @@
+using Elastic.Apm.Api;
+using Elastic.Apm.DiagnosticSource;
+using Elastic.Apm.DistributedTracing;
+using Elastic.Apm.Logging;
+using Elastic.Apm.Model;
+
+namespace Elastic.Apm.AspNetFullFramework
+{
+	public class ElasticApmModule : IHttpModule
+	{
+		private static readonly bool _isCaptureHeadersEnabled;
+		private static readonly IApmLogger _logger;
+
+		static ElasticApmModule()
+		{
+			var configReader = new FullFrameworkConfigReader(ConsoleLogger.Instance);
+			var agentComponents = new AgentComponents(configurationReader: configReader);
+			SetServiceInformation(agentComponents.Service);
+			Agent.Setup(agentComponents);
+			_logger = Agent.Instance.Logger.Scoped(nameof(ElasticApmModule));
+
+			_logger.Debug()
+				?.Log($"Entered {nameof(ElasticApmModule)} static ctor: ASP.NET: {AspNetVersion}, CLR: {ClrDescription}, IIS: {IisVersion}");
+
+			_isCaptureHeadersEnabled = Agent.Instance.ConfigurationReader.CaptureHeaders;
+
+			Agent.Instance.Subscribe(new HttpDiagnosticsSubscriber());
+		}
+
+		// We can store current transaction because each IHttpModule is used for at most one request at a time
+		// For example see https://bytes.com/topic/asp-net/answers/324305-httpmodule-multithreading-request-response-corelation
+		private Transaction _currentTransaction;
+
+		private HttpApplication _httpApp;
+
+		private static Version AspNetVersion => typeof(HttpRuntime).Assembly.GetName().Version;
+		private static string ClrDescription => RuntimeInformation.FrameworkDescription;
+		private static Version IisVersion => HttpRuntime.IISVersion;
+
+		private static void SetServiceInformation(Service service)
+		{
+			service.Framework = new Framework
+			{
+				Name = "ASP.NET",
+				Version = AspNetVersion.ToString()
+			};
+			service.Language = new Language { Name = "C#" }; //TODO
+		}
+
+		public void Init(HttpApplication httpApp)
+		{
+			_httpApp = httpApp;
+			_httpApp.BeginRequest += OnBeginRequest;
+			_httpApp.EndRequest += OnEndRequest;
+		}
+
+		public void Dispose()
+		{
+			if (_httpApp != null)
+			{
+				_httpApp.BeginRequest -= OnBeginRequest;
+				_httpApp.EndRequest -= OnEndRequest;
+				_httpApp = null;
+			}
+		}
+
+		private void OnBeginRequest(object eventSender, EventArgs eventArgs)
+		{
+			_logger.Debug()?.Log("Incoming request processing started - starting trace");
+
+			try
+			{
+				ProcessBeginRequest(eventSender, eventArgs);
+			}
+			catch (Exception ex)
+			{
+				_logger.Error()?.Log("Processing BeginRequest event failed. Exception: {Exception}", ex);
+			}
+		}
+
+		private void OnEndRequest(object eventSender, EventArgs eventArgs)
+		{
+			_logger.Debug()?.Log("Incoming request processing finished - ending trace");
+
+			try
+			{
+				ProcessEndRequest(eventSender, eventArgs);
+			}
+			catch (Exception ex)
+			{
+				_logger.Error()?.Log("Processing EndRequest event failed. Exception: {Exception}", ex);
+			}
+		}
+
+		private void ProcessBeginRequest(object eventSender, EventArgs eventArgs)
```

Comment:
nit: `eventArgs` is never used and rider complains. I think we can remove it.

Reply:
Fixed.